Source/UI.wl

```diff
@@ -978,7 +979,7 @@ promptProcess[
 						_?StringQ
 					];
 
-					<| "role" -> role, "content" -> promptCellDataToString[expr] |>
+					<| "role" -> role, "content" -> CellToString @ cell |>
 				]
```

Comment:
My only minor feedback is that instead of replacing each call to `promptCellDataToString` with `CallToString`, should continue having a dedicated function for "convert this cell to a prompt", to make it easier to e.g. conditionally decide whether to generate boxes or plain text. But that's not a big deal, we can always change this later if/when we actually want that feature.

Otherwise this LGTM.